File: app/src/main/java/com/example/imageloaderpractice/MainActivity.java
package com.example.imageloaderpractice;

import android.graphics.Bitmap;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ImageLoader imageLoader = new ImageLoader();
        //使用内存缓存
        imageLoader.setImageCache(new MemoryCache());
        //使用SD卡缓存
        imageLoader.setImageCache(new DiskCache());
        //使用双缓存
        imageLoader.setImageCache(new DoubleCache());
        //使用自定义的图片缓存实现
        imageLoader.setImageCache(new ImageCache() {
            @Override
            public Bitmap get(String url) {
                return null;
            }

            @Override
            public void put(String url, Bitmap bitmap) {

                //缓存图片；
            }
        });
    }
}

File: app/src/main/java/com/example/imageloader/MainActivity.kt
package com.example.imageloader

import android.graphics.drawable.BitmapDrawable
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.example.moduleimage.ImageLoader
import com.example.moduleimage.loaderStrategy.glide.GlideRoundedCornersTransform
import com.example.moduleimage.loaderStrategy.glide.ImageLoaderGlide
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        btn_1.setOnClickListener {
            val url = "https://tse2-mm.cn.bing.net/th/id/OIP-C.6BuL00jxYoG2oq1k17jqjwHaF5?pid=ImgDet&rs=1"
            val bitmap4 = (ContextCompat.getDrawable(this, R.mipmap.img4) as BitmapDrawable).bitmap
            ImageLoader.loadRoundedCornersImage(url, test_iv, 200f, GlideRoundedCornersTransform.CornerType.TOP_RIGHT_BOTTOM_RIGHT_BOTTOM_LEFT)
        }
    }
}